File: Pancake/Bridge/FactoryBridge.m
//
//  FactoryBridge.m
//  Pancake
//
//  Created by mxa on 14.08.2017.
//  Copyright © 2017 0bmxa. All rights reserved.
//

#import "FactoryBridge.h"
#import "Pancake-Swift.h"

// Forwards the call to the Swift implementation only.
void *Pancake_Create(CFAllocatorRef allocator, CFUUIDRef requestedTypeUUID) {
    return [PancakeFactory createWithAllocator:allocator requestedTypeUUID:requestedTypeUUID];
}

File: Pancake/Swift/FactoryBridge.swift
//
//  FactoryBridge.swift
//  Pancake
//
//  Created by mxa on 26.07.2017.
//  Copyright © 2017 mxa. All rights reserved.
//

import Foundation

@objc public class PancakeFactory: NSObject {
//    @objc public static let factory: CFPlugInFactoryFunction = { (allocator, requestedTypeUUID) -> UnsafeMutableRawPointer? in
//        let driverReference = Pancake.create(allocator: allocator, requestedTypeUUID: requestedTypeUUID)
//        return UnsafeMutableRawPointer(driverReference)
//    }
    
    @objc public static let driverInterface = Pancake.driverInterface
}
